Sources/BraintreePayPal/BTPayPalReturnURL.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+enum BTPayPalReturnURLState {
+    case unknownPath
+    case succeeded
+    case canceled
+}
+
+/// This class interprets URLs received from the PayPal app via app switch returns and web returns via ASWebAuthenticationSession.
+///
+/// PayPal app switch and ASWebAuthenticationSession authorization requests should result in success or user-initiated cancelation. These states are communicated in the url.
+struct BTPayPalReturnURL {
+
+    /// The overall status of the app switch - success, cancelation, or an unknown path
+    var state: BTPayPalReturnURLState = .unknownPath
+
+    /// Initializes a new `BTPayPalReturnURL`
+    /// - Parameter url: an incoming app switch or ASWebAuthenticationSession url
+    init?(_ redirectType: PayPalRedirectType) {
+        switch redirectType {
+        case .payPalApp(let url), .webBrowser(let url):
+            if url.path.contains("success") {
+                state = .succeeded
+            } else if url.path.contains("cancel") {
+                state = .canceled
+            } else {
+                state = .unknownPath
+            }
+        }
+    }
+
+    // MARK: - Static Methods
+
+    /// Evaluates whether the url represents a valid PayPal return URL.
+    /// - Parameter url: an app switch return URL
+    /// - Returns: `true` if the url represents a valid PayPal app switch return
```

Comment:
qq: should we also note that the `url` can be a ASWebAuthenticationSession url in the docstrings here